apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/metrics/globe.tsx

```diff
@@ -0,0 +1,112 @@
+import useResizeObserver from '@react-hook/resize-observer';
+import { Card, CardContent, CardDescription, CardTitle } from '@stackframe/stack-ui';
+import { RefObject, useEffect, useLayoutEffect, useRef, useState } from 'react';
+import Globe, { GlobeMethods } from 'react-globe.gl';
+
+function useSize(target: RefObject<HTMLDivElement>) {
+  const [size, setSize] = useState<DOMRectReadOnly>();
+
+  useLayoutEffect(() => {
+    if (!target.current) return;
+    setSize(target.current.getBoundingClientRect());
+  }, [target]);
+
+  // Where the magic happens
+  useResizeObserver(target, (entry) => setSize(entry.contentRect));
+  return size;
+}
+
+const ringData = [
+  { lat: 37.792080, lng: -122.432361 }
+];
+
+
+const PLACEHOLDER_COUNTRY_DATA: Record<string, number> = {
+  'US': 123,
+  'CA': 45,
+  'JP': 67,
+  'KR': 89,
+};
+
+export function GlobeSection({ countryData, children }: {countryData: Record<string, number>, children?: React.ReactNode}) {
+  const globeRef = useRef<GlobeMethods>();
+  const ref = useRef<HTMLDivElement>(null);
+  const size = useSize(ref);
+
+  const [countries, setCountries] = useState<any>({ features: []});
+  const [selectedCountry, setSelectedCountry] = useState<string|null>(null);
+  const [selectedCountryName, setSelectedCountryName] = useState<string|null>(null);
+
+  const maxCountryCount = Math.max(0, ...Object.values(countryData));
+
+  useEffect(() => {
+    // load data
+    fetch('/static/country-data.geojson')
+      .then(res => res.json())
+      .then(setCountries)
+      .catch(e => { throw e; });
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    let cancelled = false;
    
    runAsynchronously(async () => {
      const fetchRes = await fetch('/static/country-data.geojson');
      const json = fetchRes.json();
      if (!cancelled) setCountries(json);
    });
    
    return () => cancelled = true;
  }, []);
```

(or react-query, if you prefer)